stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -83,4 +125,49 @@ public OutputStream openResponseBodyFile(String requestId, boolean base64Encode)
   private static String getFilename(String requestId) {
     return FILENAME_PREFIX + requestId;
   }
+
+  /**
+   * Associates an asynchronous pretty printer with a response request id
+   * The pretty printer will be used to pretty print the response body that has
+   * the particular request id
+   *
+   * @param requestId Unique identifier for the response
+   * as per {@link NetworkEventReporter.InspectorResponse#requestId()}
+   * @param asyncPrettyPrinter Asynchronous Pretty Printer to pretty print the response body
+   */
+  public void associateAsyncPrettyPrinterWithId(
+      String requestId,
+      AsyncPrettyPrinter asyncPrettyPrinter) {
+    if (mRequestIdMap.containsKey(requestId)) {
+      throw new IllegalArgumentException("cannot associate different " +
+          "pretty printers with the same request id: "+requestId);
+    }
+    mRequestIdMap.put(requestId, asyncPrettyPrinter);
+  }
+
+  private class AsyncPrettyPrintingCallable implements Callable<String> {
+    private final InputStream mInputStream;
+    private final AsyncPrettyPrinter mAsyncPrettyPrinter;
+
+    public AsyncPrettyPrintingCallable(
+        InputStream in,
+        AsyncPrettyPrinter asyncPrettyPrinter) {
+      mInputStream = in;
+      mAsyncPrettyPrinter = asyncPrettyPrinter;
+    }
+
+    @Override
+    public String call() throws IOException {
+      return prettyPrintContent(mInputStream, mAsyncPrettyPrinter);
+    }
+
+    private String prettyPrintContent(InputStream in, AsyncPrettyPrinter asyncPrettyPrinter)
+        throws IOException {
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      PrintWriter writer = new PrintWriter(out);
+      asyncPrettyPrinter.printTo(writer, in);
```

Comment:
My bad, I just noticed you have timeout logic above, but with more overhead than is needed.  A handler can do this cheaper than an executor, though I don't require that you use a handler in this case since it occurs only once per network request that is to be pretty printed.